alertmanager/rootfs/boot.go

```diff
@@ -0,0 +1,54 @@
+package main
+
+import (
+	"errors"
+	"log"
+	"os"
+	"strconv"
+	"time"
+
+	"golang.org/x/net/context"
+
+	"github.com/coreos/etcd/client"
+)
+
+func main() {
+	var etcdTTL int
+	var etcdPort string
+
+	if os.Getenv("ETCD_PORT") == "" {
+		etcdPort = "4001"
+	} else {
+		etcdPort = os.Getenv("ETCD_PORT")
+	}
+
+	if os.Getenv("ETCD_TTL") == "" {
+		etcdTTL = 20
+	} else {
+		etcdTTL, _ = strconv.Atoi(os.Getenv("ETCD_TTL"))
+	}
+
+	cfg := client.Config{
+		Endpoints: []string{"http://" + os.Getenv("HOST") + ":" + etcdPort},
+		Transport: client.DefaultTransport,
+		// set timeout per request to fail fast when the target endpoint is unavailable
+		HeaderTimeoutPerRequest: time.Duration(time.Second),
+	}
+
+	c, err := client.New(cfg)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	kapi := client.NewKeysAPI(c)
+
+	ctx, _ := context.WithTimeout(context.TODO(), 100*time.Millisecond)
```

Comment:
@lanej you're welcome. I'm open for questions whenever. Also, `context.Context` is not an official part of `net/http`, but it's very helpful.